sympy/concrete/tests/test_products.py

```diff
@@ -234,6 +234,12 @@ def test__eval_product():
     assert product(2*a(i), (i, 1, n)) == 2**n * Product(a(i), (i, 1, n))
     # issue 4810
     assert product(2**i, (i, 1, n)) == 2**(n/2 + n**2/2)
+    k, m = symbols('k m', integer=True)
+    assert product(2**i, (i, k, m)) == 2**(-k**2/2 + k/2 + m**2/2 + m/2)
+    n = Symbol('n', negative=True, integer=True)
+    p = Symbol('p', positive=True, integer=True)
+    assert product(2**i, (i, n, p)) == 2**(-n**2/2 + n/2 + p**2/2 + p/2)
+    assert product(2**i, (i, p, n)) == 2**(n**2/2 + n/2 - p**2/2 + p/2)
```

Comment:
How is the product defined when the end index is less than the start? Is it just the reciprocal of the product with limits flipped? Does something similar happen for sums?

Reply:
Products:
https://github.com/sympy/sympy/blob/a387ec651951f72b77ca45b3a43c0a45bc142dfe/sympy/concrete/products.py#L210-L214

Sums:
https://github.com/sympy/sympy/blob/a387ec651951f72b77ca45b3a43c0a45bc142dfe/sympy/concrete/summations.py#L190-L194

I do not fully understand it to be honest. The change here (motivating the test), was that it currently only swapped with numerical limits. Not sure that the current test actually tests it, but at least it should exercise the code in a new way (it is the same result as earlier).